packages/ragbits-core/src/ragbits/core/embeddings/vertex_multimodal.py

```diff
@@ -0,0 +1,163 @@
+        """
+        if not HAS_LITELLM:
+            raise ImportError("You need to install the 'litellm' extra requirements to use LiteLLM embeddings models")
+
+        super().__init__()
+        if model.startswith(self.VERTEX_AI_PREFIX):
+            model = model[len(self.VERTEX_AI_PREFIX) :]
+
+        self.model = model
+        self.api_base = api_base
+        self.api_key = api_key
+        self.concurency = concurency
+        self.options = options or {}
+
+        supported_models = VertexMultimodalEmbedding().SUPPORTED_MULTIMODAL_EMBEDDING_MODELS
+        if model not in supported_models:
+            raise ValueError(f"Model {model} is not supported by VertexAI multimodal embeddings")
+
+    async def _embed(self, data: list[dict]) -> list[dict]:
+        """
+        Creates embeddings for the given data. The format is defined in the VertexAI API:
+        https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-multimodal-embeddings
+
+        Args:
+            data: List of instances in the format expected by the VertexAI API.
+
+        Returns:
+            List of embeddings for the given VertexAI instances, each instance is a dictionary
+            in the format returned by the VertexAI API.
+
+        Raises:
+            EmbeddingStatusError: If the embedding API returns an error status code.
+            EmbeddingResponseError: If the embedding API response is invalid.
+        """
+        semaphore = asyncio.Semaphore(self.concurency)
+        try:
+            response = await asyncio.gather(
+                *[self._call_litellm(instance, semaphore) for instance in data],
+            )
+        except VertexAIError as exc:
+            raise EmbeddingStatusError(exc.message, exc.status_code) from exc
+
+        embeddings: list[dict] = []
+        for i, embedding in enumerate(response):
+            if embedding.data is None or not embedding.data:
+                raise EmbeddingResponseError(f"No embeddings returned for instance {i}")
+            embeddings.append(embedding.data[0])
+
+        return embeddings
+
+    async def _call_litellm(self, instance: dict, semaphore: asyncio.Semaphore) -> litellm.EmbeddingResponse:
+        """
+        Calls the LiteLLM API to get embeddings for the given data.
+
+        Args:
+            instance: Single VertexAI instance to get embeddings for.
+            semaphore: Semaphore to limit the number of concurrent requests.
+
+        Returns:
+            List of embeddings for the given LiteLLM instances.
+        """
+        async with semaphore:
+            response = await litellm.aembedding(
+                input=[instance],
+                model=f"{self.VERTEX_AI_PREFIX}{self.model}",
+                api_base=self.api_base,
+                api_key=self.api_key,
+                **self.options,
+            )
+
+        return response
+
+    async def embed_text(self, data: list[str]) -> list[list[float]]:
```

Comment:
We've merged audit module, this method and `embed_image` should be decorated with `@traceable` or use the `trace()` context manager to log inputs and outputs. 

Ref implementation for [`LiteLLMEmbeddings`](https://github.com/deepsense-ai/ragbits/blob/e674abc8b6837cb2fba3f135c27755c5d7955908/packages/ragbits-core/src/ragbits/core/embeddings/litellm.py#L73-L105)

Reply:
Does this make sense?

Reply:
Looks good, I'd remove the `concurency` because we will see it in the traces

Reply:
Ok, removed